app/(user)/profile/settings/components/PasswordSection.js

```diff
@@ -0,0 +1,177 @@
+            oldPassword,
+        };
+        const response = await gpFetch(api, payload, 3600);
+        const { user } = response;
+        setUserCookie(user);
+        // yield put(snackbarActions.showSnakbarAction('Your new password is saved'));
+        // const random = parseInt(Math.random() * 1000 + '', 10);
+        // yield put(push(`?save=${random}`)); // force a refresh
+    } catch (error) {
+        console.log('error', error);
+        // if (error.response?.incorrect) {
+        //     yield put(
+        //         snackbarActions.showSnakbarAction(
+        //             'Current Password is incorrect',
+        //             'error',
+        //         ),
+        //     );
+        //     logEvent('change-password', 'error - incorrect password');
+        //     yield put(
+        //         globalActions.logErrorAction(
+        //             'change password - incorrect password',
+        //             error,
+        //         ),
+        //     );
+        // } else {
+        //     yield put(
+        //         snackbarActions.showSnakbarAction(
+        //             'Error changing your password.',
+        //             'error',
+        //         ),
+        //     );
+        //     logEvent('change-password', 'error');
+        // }
+    }
+}
+function PasswordSection() {
+    const userState = useHookstate(globalUserState);
+    const user = userState.get();
+
+    const initialState = {
+        oldPassword: '',
+        password: '',
+    };
+    const [state, setState] = useState(initialState);
+
+    const onChangeField = (key, val) => {
+        setState({
+        ...state,
+        [key]: val,
+        });
+    };
+
+    const canSave = () => {
+        if (
+            user.hasPassword &&
+            state.password !== '' &&
+            state.oldPassword !== '' &&
+            state.password.match(passwordRegex) &&
+            state.password.length > 7
+        ) {
+            return true;
+        }
+        if (
+            !user.hasPassword &&
+            state.password !== '' &&
+            state.password.length > 7
+        ) {
+            return true;
+        }
+        return false;
+    };
+
+    const reset = () => {
+        setState(initialState);
+    };
+
+    const handleSavePassword = () => {
+        if (canSave()) {
+            changePasswordCallback(state.password, state.oldPassword);
+            reset();
+        }
+    };
+
+    return (
+        <section>
+            <PortalPanel color="#CA2CCD">
+                <form noValidate onSubmit={(e) => e.preventDefault()}>
+                    <h3 
+                        className="text-[22px] tracking-wide font-black mb-16" data-cy="settings-title"
+                    >
+                        {user.hasPassword ? 'Change' : 'Create'} your password
+                    </h3>
+                    <div className="grid grid-cols-12 gap-3">
+                        <div className="col-span-12 md:col-span-6">
```

Comment:
this can be just
```
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-3">
                        <div>

```

but I guess both ways work. 
I would like us to be consistent in the way we use the grid.

Reply:
can you change yours to match mine?